src/v/redpanda/application.cc

```diff
@@ -1556,8 +1560,18 @@ void application::wire_up_redpanda_services(
             }))
           .get();
         construct_service(
-          cloud_storage_api,
+          cloud_io,
           std::ref(cloud_storage_clients),
+          ss::sharded_parameter(
+            [&cloud_configs] { return cloud_configs.local().client_config; }),
```

Comment:
can we avoid depending on archival storage? maybe we can't do that easily yet until we clean other stuff up?

Reply:
Yea I think we could probably update the conditions to include usage of shadow topics or iceberg. For now though it doesn't seem like there's a good reason to initialize without cloud storage enabled.